darts/models/regression_ensemble_model.py

```diff
@@ -0,0 +1,100 @@
+
+try:
+    from darts.models.torch_forecasting_model import TorchForecastingModel
+    TORCH_AVAILABLE = True
+except ImportError:
+    logger.warning('Torch not available')
+    TORCH_AVAILABLE = False
+
+
+class RegressionEnsembleModel(EnsembleModel):
+    def __init__(self,
+                 forecasting_models: List[ForecastingModel],
+                 regression_train_n_points: int,
+                 regression_model=LinearRegression(n_jobs=-1, fit_intercept=False)):
+        """
+        Class for ensemble models using a regression model for ensembling individual models' predictions
+        The provided regression model must implement fit() and predict() methods
+        (e.g. scikit-learn regression models)
+
+        Parameters
+        ----------
+        forecasting_models
+            List of forecasting models whose predictions to ensemble
+        regression_train_n_points
+            The number of points to use to train the regression model
+        regression_model
+            Any regression model with predict() and fit() methods (e.g. from scikit-learn)
+            Default: `sklearn.linear_model.LinearRegression(n_jobs=-1, fit_intercept=False)`
+        """
+        super().__init__(forecasting_models)
+
+        # wrap provided regression_model in a StandardRegressionModel (if not already the case)
+        if isinstance(regression_model, StandardRegressionModel):
+            # raise exception if train_n_points value is ambiguous
+            model_train_n_points = regression_model.train_n_points
+            raise_if(model_train_n_points is not None and regression_train_n_points != model_train_n_points,
+                     "Provided StandardRegressionModel.train_n_points parameter doesn't match specified"
+                     " regression_train_n_points parameter.",
+                     logger)
+
+            regression_model.train_n_points = regression_train_n_points
+        else:
+            regression_model = StandardRegressionModel(regression_train_n_points, regression_model)
+
+        self.regression_model = regression_model
+
+    def fit(self, training_series: TimeSeries, target_series: Optional[TimeSeries] = None) -> None:
+        # TODO: Factor this out, same logic is used in 3 places already. Need to find an appropiate name
+        if target_series is None:
+            target_series = training_series
+        raise_if_not(all(training_series.time_index() == target_series.time_index()),
+                     "training and target series must have same time indices.",
+                     logger)
+
+        # spare train_n_points points to serve as regression target
+        raise_if(len(training_series) <= self.regression_model.train_n_points,
+                 "regression_train_n_points parameter too big (greater or equal"
+                 " the number of points in training_series)",
+                 logger)
+        forecast_training = training_series[:-self.regression_model.train_n_points]
```

Comment:
It would be nice to rely on the new `historical_forecasts()` method here, and make it return the time series of historical forecasts, which you would then use as features. wdyt?

Reply:
This got me thinking quite a bit :)

I like the idea, but the implementation might be tricky and may require to either make quite a few assumptions for the user, e.g. which values to use for `stride`, `forecast_horizon`, `retrain`, … or complicate the use of `RegressionEnsembleModel` quite a bit.

And even then it would still make one pretty big assumption if we let `last_points_only=True`: the regression would only be trained on (and thus only care about) the last points of the models' forecasts, rendering the class almost useless if you care about the whole forecast horizon.

This made me think that the same thing can be said about the current implementation though, it’s gives weight to every point in the forecast horizon, so `StandardRegressionModel` possibly isn’t as good as it could be if you only care about the last point of a prediction. This doesn't seem as bad to me though, as the regression learns which model performs "best" overall but you can still use it only caring about the last point predictions it makes.

To keep it nice and simple **I’d be for leaving it as such (at least for now)**

Nevertheless I think it could be doable and useful to latter have maybe another class which would use `historical_forecasts()` and a regression model, that would take pretty much the same parameters as `historical_forecasts()`.

It could handle the case of `last_points_only=False` by enforcing `forecast_horizon = stride` such that we can still create a single time series per model of the historical forecasts, and use these as feature time series to train the regression model. The behavior of the current `RegressionEnsembleModel` above would then be a special case of this model, with `forecast_horizon = regression_train_n_points`, `start=len(series)-forecast_horizon` and `last_points_only=False`.

Reply:
OK makes sense, agree, we can leave it as is at least for now :)